source/material_model/visco_plastic.cc

```diff
@@ -133,6 +140,50 @@ namespace aspect
     evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
              MaterialModel::MaterialModelOutputs<dim> &out) const
     {
+      if (enable_material_lookup) {
+      std::vector<std::vector<double>> volume_fractions_lookup(in.n_evaluation_points(), std::vector<double>(equation_of_state_lookup.number_of_lookups()));
+
+      std::vector<EquationOfStateOutputs<dim>> eos_outputs_lookup (in.n_evaluation_points(), equation_of_state_lookup.number_of_lookups());
+      // We need to make a copy of the material model inputs because we want to use the adiabatic pressure
+      // rather than the real pressure for the equations of state (to avoid numerical instabilities).
+      MaterialModel::MaterialModelInputs<dim> eos_in_lookup(in);
+      for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
+        eos_in_lookup.pressure[i] = this->get_adiabatic_conditions().pressure(in.position[i]);
+
+      // Evaluate the equation of state properties over all evaluation points
+      equation_of_state_lookup.evaluate(eos_in_lookup, eos_outputs_lookup);
+      
+      for (unsigned int i = 0; i < in.n_evaluation_points(); ++i)
+        {
+          // Calculate volume fractions from mass fractions
+          // If there is only one lookup table, set the mass and volume fractions to 1
+          std::vector<double> mass_fractions;
+          if (equation_of_state_lookup.number_of_lookups() == 1)
+            mass_fractions.push_back(1.0);
+          else
+            {
+              // We only want to compute mass/volume fractions for fields that are chemical compositions.
+              std::vector<double> chemical_compositions;
+              const std::vector<CompositionalFieldDescription> composition_descriptions = this->introspection().get_composition_descriptions();
+
+              for (unsigned int c = 0; c < in.composition[i].size(); ++c)
+                if (composition_descriptions[c].type == CompositionalFieldDescription::chemical_composition || composition_descriptions[c].type == CompositionalFieldDescription::unspecified)
+                  chemical_compositions.push_back(in.composition[i][c]);
+
+              mass_fractions = MaterialUtilities::compute_composition_fractions(chemical_compositions, *composition_mask);
```

Comment:
This should be simpler since #5039  was merged, you can do:
```suggestion
              mass_fractions = MaterialUtilities::compute_composition_fractions(this->introspection.chemical_composition_field_indices(), *composition_mask);
```

This requires you do specify the compositional field types input parameter for your model, but that will be required in the future anyway and since you are working on a new feature you might as well start with it.